packages/codegen-ui-react/lib/forms/react-form-renderer.ts

```diff
@@ -0,0 +1,324 @@
+    componentText += propsPrinted;
+
+    const result = printer.printNode(EmitHint.Unspecified, wrappedFunction, file);
+    componentText += result;
+
+    const { componentText: transpiledComponentText, declaration } = transpile(componentText, this.renderConfig);
+
+    return {
+      componentText: transpiledComponentText,
+      declaration,
+      renderComponentToFilesystem: async (outputPath: string) => {
+        await this.renderComponentToFilesystem(transpiledComponentText)(this.fileName)(outputPath);
+        if (declaration) {
+          await this.renderComponentToFilesystem(declaration)(getDeclarationFilename(this.fileName))(outputPath);
+        }
+      },
+    };
+  }
+
+  renderFunctionWrapper(
+    componentName: string,
+    variableStatements: Statement[],
+    jsx: JsxElement | JsxFragment | JsxSelfClosingElement,
+    renderExport: boolean,
+  ): FunctionDeclaration {
+    const componentPropType = getComponentPropName(componentName);
+    const jsxStatement = factory.createReturnStatement(
+      factory.createParenthesizedExpression(
+        this.renderConfig.script !== ScriptKind.TSX
+          ? jsx
+          : /* add ts-ignore comment above jsx statement. Generated props are incompatible with amplify-ui props */
+            addSyntheticLeadingComment(
+              factory.createParenthesizedExpression(jsx),
+              SyntaxKind.MultiLineCommentTrivia,
+              ' @ts-ignore: TS2322 ',
+              true,
+            ),
+      ),
+    );
+    const codeBlockContent = variableStatements.concat([jsxStatement]);
+    const modifiers: Modifier[] = renderExport
+      ? [factory.createModifier(SyntaxKind.ExportKeyword), factory.createModifier(SyntaxKind.DefaultKeyword)]
+      : [];
+    const typeParameter = PrimitiveTypeParameter[Primitive[this.formComponent?.componentType as Primitive]];
+    // only use type parameter reference if one was declared
+    const typeParameterReference = typeParameter && typeParameter.declaration() ? typeParameter.reference() : undefined;
+    return factory.createFunctionDeclaration(
+      undefined,
+      modifiers,
+      undefined,
+      factory.createIdentifier(componentName),
+      typeParameter ? typeParameter.declaration() : undefined,
+      [
+        factory.createParameterDeclaration(
+          undefined,
+          undefined,
+          undefined,
+          'props',
+          undefined,
+          factory.createTypeReferenceNode(componentPropType, typeParameterReference),
+          undefined,
+        ),
+      ],
+      factory.createTypeReferenceNode(
+        factory.createQualifiedName(factory.createIdentifier('React'), factory.createIdentifier('ReactElement')),
+        undefined,
+      ),
+      factory.createBlock(codeBlockContent, true),
+    );
+  }
+
+  abstract renderJsx(component: StudioComponent, parent?: StudioNode): JsxElement | JsxFragment | JsxSelfClosingElement;
```

Comment:
should this abstract member be so far down in the class?

Reply:
Following the layout:
1. public
2. abstract
3. private functions